### PR TITLE
Removed section related to access denied 

### PR DESCRIPTION
**Pull request by srinivasanagandla-okta**
## Description:
- Primary authentication no longer returns 403 when access is denied based on sign-on policy
- Removed the section about 403 during primary authentication and added a note about it

## Type of Pull Request:

- [ ] Blog Post
- [X] Content (Documentation updates or typo-fixes)
- [ ] Functional (CSS changes, JS updates, or layout modifications)

### Resolves:
* [OKTA-123540](https://oktainc.atlassian.net/browse/OKTA-123540)

## How Has This Been Tested?
- [X] I have built this locally and verified that it does not break existing functionality.
- [ ] For functional changes, I have tested the change across Chrome, Firefox, Safari, and mobile.

## Primary Reviewers:
@mystiberry-okta @tthompson-okta @edjohns-okta 

## Before Merging:

- [ ] I have had this PR reviewed and approved on GitHub.
- [ ] I have squashed my commits into `1` commit or will use the **Squash and Merge button**.

### CODE REVIEW

**Comment by rchild-okta** [comment]
@srinivasanagandla-okta couple points now that we're moving to a different model (see @mystiberry-okta's email from earlier today):

- The PR needs to be based against the `weekly` branch. This is the new branch we're using for commits that are going out with the next weekly release.
- I noticed in your PR comment that you didn't update the jira number.

**Comment by srinivasanagandla-okta** [comment]
@rchild-okta Thanks. I updated the PR based on your comments.

File: _source/_docs/api/resources/authn.md [hunk 1]
```diff
@@ -107,6 +107,8 @@ It is recommended that you generate a UUID or GUID for each client and persist t
 
 `401 Unauthorized` status code is returned for requests with invalid credentials or locked out accounts.
 
+> Primary authentication could also fail with 401 when access is denied based on sign-on policy.
```
Comment (tthompson-okta) [inline _source/_docs/api/resources/authn.md]: Nit: we should be consistent with our references to "401" and other HTTP codes; in the existing doc it's highlighted with `401 Unauthorized`; seems like we should do that here too.

Wording-wise, I would suggesting not using the world "fail" here, and maybe just merge with the sentence above. For example:
```
'401 Unauthorized' status code is returned for requests with invalid credentials, locked out accounts, or access denied by sign-on policy.
```

Comment (srinivasanagandla-okta) [inline _source/_docs/api/resources/authn.md]: I like your idea of merging this with the existing sentence. @mystiberry-okta let me know your thought too on that.

Comment (mystiberry-okta) [inline _source/_docs/api/resources/authn.md]: agree!

**Comment by srinivasanagandla-okta** [comment]
@tthompson-okta @mystiberry-okta I have updated the PR based on the feedback. Please review.

**Pull request merged**